### DISCUSSION TITLE
Donation Dialog

### DISCUSSION THREAD

**Discussion by Jeffser** [Jeffser/Alpaca#176] [Polls]
Hi, I added a donation dialog to the app for version 1.0.3, it looks like this
![image](https://github.com/user-attachments/assets/307c0cda-f5ce-4421-ad96-0ac845176e76)

This dialog appears randomly in a 1 to 100 change so it's kinda rare, the user can also choose to not show the dialog again.

When the user presses the 'don't show again' or 'support' option the dialog will stop showing up on that installation of Alpaca.

I believe that open source projects should at least listen to the opinion of the users so I wanted to ask you all if this is fine by your standards or if I should remove it for the next version.

Update:

It was suggested to me that I change the sponsor dialog for a link inside the about dialog as part of my submission to Gnome Circle. Nothing against that decision, it makes sense that apps from that initiative shouldn't be begging for money lol

Seriously tho, I'm fine with removing the dialog, I should've asked them before adding it in the first place.

https://gitlab.gnome.org/Teams/Circle/-/issues/230#note_2186581

https://github.com/Jeffser/Alpaca/commit/378689be394e690afa55fa98708c0ec1bde7c85e

https://github.com/Jeffser/Alpaca/commit/6e9ddfcbf2303c3f1da94518615a22a7692cccd9

**Comment by Jeffser** [comment]
I don't want people to think I'm feeling entitled to donations

Alpaca is open source and free to use but I could use some money, that's all

I will do whatever the votes say in a couple of weeks

**Reply by olumolu** [reply]
I would say the dialogue should be smaller in size so it looks good. Else this has no issues.

**Comment by Luiz-C-Lima** [comment]
I think that 1 in 100 is a bit too rare, I have opened Alpaca 1.0.3 several times and have never seen it, maybe 1 in 50 or even 30, would be better, most people are not going to open Alpaca more than twice in a day, right now it could take months for a user to see this dialog, and the user has the option to tell it to not show it again anyway, so it would not be bothersome.

I think that a donation button should also be added either to the hamburger menu or the about section.

**Reply by Jeffser** [reply]
alright, I changed it to 1 in 50 change

**Reply by Luiz-C-Lima** [reply]
Could you update the pot file for that? I have finished the translation, but the donation dialog is not available yet.

**Reply by Jeffser** [reply]
on it

**Reply by Jeffser** [reply]
[Ready!](https://github.com/Jeffser/Alpaca/commit/acdce762c9be850f99104b8fe5fe131a9849193c)

thanks

**Comment by GaPhilippone** [comment]
Hello, if i can i want to support when i got money.
One thing that would be great is to put in the info about the app, an option for donation especially when a user click "Don't show again" not because he can't donate right now but is annoyed from the popup BUT still want to donate when it is possible for him.

**Reply by Jeffser** [reply]
Don't worry about donating if you don't have the means.

I will add a toast message (those small notifications at the bottom of the app) that will say something like "Visit the GitHub if you change your mind" or something like that

**Reply by Jeffser** [reply]
This
![image](https://github.com/user-attachments/assets/65410780-14f8-4633-8cea-2b64e81f172f)

**Comment by Izaic** [comment]
You're putting time into it, and time is valuable, so if someone wants to donate to support the project and say thanks that's perfectly fine. Heck, I don't think most people would even be mad if you had the popup come up 100% of the time for the first launch of the application.

**Reply by Jeffser** [reply]
Thanks for the kind words, I think people should at least try it before the popup showing up so it's impossible for it to appear on the first launch.
